pandas/core/algorithms.py

```diff
@@ -268,69 +267,81 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
 
     if com.is_categorical_dtype(values.dtype):
```

Comment:
actually with only a small rewriting I think you could do:

```
if com.is_extension_dtype(values):
    result = values.values_counts(dropna)
    result.name = name
else:
   .....
```

if we define ``DateTimeIndex.value_counts`` that properly handles localization before/after (instead of in here)
will be much cleaner.

Further it makes sense to define ``.value_counts()`` in ``tseries/base.py`` (and move the inference from ``core/base.py/value_counts.py`` there. To handle the dispatching of period/DatetimeIndex.

a bit of refactoring but moves functionaility where it should be.